### PR TITLE
ci: add workflow_dispatch inputs for manual image tag override

### LINKED ISSUE #489: Add workflow_dispatch inputs for publish.yml (image tag override)
The manual `workflow_dispatch` trigger added in #488 fires with no inputs, which means it uses only the metadata-action defaults (semver tags from the git ref). For manual re-runs (e.g. hotfix without a new tag), an operator has no way to specify a custom image tag. Adding an optional `image_tag` string input to `workflow_dispatch` would make manual publishes more flexible without breaking the tag-triggered path.

_Follow-up from #336_

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary
Enable operators to manually specify a custom image tag when triggering the publish workflow, improving flexibility for hotfix releases without requiring a new git tag.

## Changes
- Add optional `image_tag` string input to `workflow_dispatch` trigger in publish.yml
- Implement build-tags.sh script to generate image tags from either git ref metadata or manual override input
- Add test-build-tags.sh to verify tag generation logic across different scenarios

## Testing
- Verify manual workflow_dispatch with custom image_tag produces expected tag
- Verify tag-triggered workflow (no manual input) uses semver from git ref
- Run test-build-tags.sh to confirm tag generation logic covers edge cases

## Closes
Closes #489

Generated by Claude Code via ProjectHephaestus automation.

### CODE REVIEW

File: .github/workflows/scripts/build-tags.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+stripped="${stripped//$'\n'/}"
+stripped="${stripped//$'\t'/}"
+
+# Keep only the Docker tag charset: [A-Za-z0-9_.-]. Anything else becomes ''.
+clean="$(printf '%s' "$stripped" | LC_ALL=C tr -cd 'A-Za-z0-9_.-')"
+
+# Docker rejects leading '.' or '-'; strip until the first valid char.
+clean="${clean##[.-]}"
+
+# Truncate to Docker's 128-char tag limit.
```
Comment (mvillmow) [inline .github/workflows/scripts/build-tags.sh:29]: The comment says "strip until the first valid char," but `${clean##[.-]}` removes only a single leading `.`/`-` (the `[.-]` pattern matches exactly one char). Input like `.-foo` yields `-foo`, which Docker rejects as a tag (no leading `-`/`.`). Use a loop or extglob to strip the full leading run, e.g. `clean="${clean#"${clean%%[!.-]*}"}"`, or update the comment to reflect that only one separator is stripped (relying on docker/metadata-action to re-sanitize).

**Review by mvillmow — COMMENTED** [review]
Correct and complete; plan's e2e step-ordering bug is fixed. Minor: leading-separator strip removes only one char despite comment claiming full strip.

File: coverage.xml [hunk 2]
```diff
@@ -0,0 +1,855 @@
+<?xml version="1.0" ?>
```
Comment (mvillmow) [inline coverage.xml]: This generated coverage report (855 lines) is unrelated to the issue (#489 workflow_dispatch image_tag) and should not be committed. It embeds an absolute machine-specific path (`/home/mvillmow/Projects/ProjectHermes/build/.worktrees/issue-489`) in the `<source>` element, leaking a local filesystem layout. Remove it from the PR and add `coverage.xml` to `.gitignore`.

**Review by mvillmow — COMMENTED** [review]
Workflow change is correct; blocked by accidental commit of generated coverage.xml (absolute local path, unrelated to issue).
